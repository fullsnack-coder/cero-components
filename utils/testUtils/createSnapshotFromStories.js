import React from 'react'
import { render } from '@testing-library/react'
import { composeStories } from '@storybook/testing-react'

// TODO: Move this to a generic approach where it
// reads all the stories and works as Storyshoot
const createSnapshotFromStories = (componentName, stories) => {
  const composedStories = composeStories(stories)

  Object.entries(composedStories).forEach(([story, Component]) => {
    describe(`when '${componentName}' ${story} is mounted`, () => {
      it('should render', () => {
        const { asFragment } = render(<Component />)
        expect(asFragment()).toMatchSnapshot()
      })
    })
  })
}

export default createSnapshotFromStories
