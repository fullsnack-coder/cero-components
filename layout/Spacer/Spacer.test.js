import * as stories from './Spacer.stories'
import createSnapshotFromStories from '../../utils/testUtils/createSnapshotFromStories'

describe('[ layout / Spacer ]', () => {
  describe('#Spacer', () => createSnapshotFromStories('Spacer', stories))
})
