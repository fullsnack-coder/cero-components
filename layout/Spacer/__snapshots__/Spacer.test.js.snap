// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[ layout / Spacer ] #Spacer when 'Spacer' Default is mounted should render 1`] = `
<DocumentFragment>
  <div
    class="spacer is-playground"
    style="display: inline-block; width: 0px; height: 0px;"
  />
</DocumentFragment>
`;

exports[`[ layout / Spacer ] #Spacer when 'Spacer' Sizes is mounted should render 1`] = `
<DocumentFragment>
  <div
    class="spacer is-playground"
    style="display: inline-block; width: 0px; height: 0px;"
  />
  <div
    class="spacer is-playground"
    style="display: inline-block; width: 5px; height: 5px;"
  />
  <div
    class="spacer is-playground"
    style="display: inline-block; width: 15px; height: 15px;"
  />
  <div
    class="spacer is-playground"
    style="display: inline-block; width: 20px; height: 20px;"
  />
  <div
    class="spacer is-playground"
    style="display: inline-block; width: 40px; height: 40px;"
  />
</DocumentFragment>
`;
